### PR TITLE
Move Examples to Tests

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
Following #1693, I removed most examples in the `examples/` folder by:
- Turning important examples into integration tests
- Merging relevant information that was not explained elsewhere into the docs

### Not-Removed Examples
The following examples were *not* removed/updated:
- **quickstart:** must be kept for obvious reasons
- **label studio:** Well be tackled separately in #1628 
- **generative chat:** will also be tackled in a separate PR

### Major Code Changes
- The `zenml example ...` CLI commands were removed.

### Major Docs Changes
- Added materializer overview tables: https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/vlf6HVLGPRknfyMwMtR2/user-guide/advanced-guide/artifact-management/handle-custom-data-types
- Added Facets as an example into the artifact visualization page: https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/vlf6HVLGPRknfyMwMtR2/user-guide/advanced-guide/artifact-management/visualize-artifacts#example-facets-data-skew-visualization
- Majorly updated many component guide pages with infos and screenshots from old READMEs: https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/vlf6HVLGPRknfyMwMtR2/user-guide/component-guide

#### Orchestrators
- **Kubeflow** -> turned into `tensorflow` integration test example
- **Kubernetes** -> turned into `sklearn` integration test example; updated component guide
- **Other** (Airflow, Vertex, Tekton, Sagemaker) -> deleted and updated component guide

#### Experiment Trackers, Model Registries, Model Deployers
- **MLflow** (tracking, registry, deployment) -> merged into one "mlflow" example with three separate integration test cases. Also majorly updated component guide.
- **Other Trackers** (Neptune, WandB) -> deleted (since not automatically testable and redundant once experiment tracker abstraction will be implemented) and updated component guide
- **BentoML** -> moved to tests (but no automatically runnable) and updated component guide
- **Kserve** -> deleted since deprecated
- **Seldon + Custom Deployment** -> merged into one example, moved to tests, updated component guide

#### Materializers
- **Facets** -> moved to tests and added to artifact visualization docs page: https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/vlf6HVLGPRknfyMwMtR2/user-guide/advanced-guide/artifact-management/visualize-artifacts#example-facets-data-skew-visualization
- **Other** (Scipy, PyTorch, Neural Prophet, LightGBM, XGB) -> moved to tests

#### Other
- **Data Validators** (Deepchecks, Evidently, GE, Whylogs) -> moved to tests
- **Step Operators** (Spark, step_operator_remote_training) -> deleted and updated component guide
- **Slack** -> moved to tests (but not automatically runnable) and updated component guide
- **Feast** -> deleted and updated component guide
- **Other** (adjust_your_own, dynamic pipelines, hooks, secrets, code_repository) -> deleted

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: docs/book/user-guide/component-guide/experiment-trackers/mlflow.md [hunk 1]
```diff
@@ -195,7 +195,54 @@ def tf_trainer(
...
+    ...
+```
+{% endhint %}
+
+### MLflow UI
+
+MLflow comes with its own UI that you can use to find further details about 
+your tracked experiments.
+
+you can find the URL of the MLflow experiment linked to a specific ZenML run
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/experiment-trackers/mlflow.md:219]:
```suggestion
You can find the URL of the MLflow experiment linked to a specific ZenML run 
```

File: docs/book/user-guide/component-guide/experiment-trackers/neptune.md [hunk 2]
```diff
@@ -145,8 +147,49 @@ def tf_trainer(
...
+def tf_trainer(...):
     ...
 ```
+{% endhint %}
+
+### Neptune UI
+
+Neptune comes with a web-based UI that you can use to find further details about 
+your tracked experiments. Each pipeline run will be logged as a separate 
+experiment run in Neptune, which you can inspect in the Neptune UI:
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/experiment-trackers/neptune.md:175]:
```suggestion
experiment run in Neptune, which you can inspect on the Neptune UI:
```

File: docs/book/user-guide/component-guide/experiment-trackers/wandb.md [hunk 3]
```diff
@@ -145,8 +145,60 @@ def tf_trainer(
...
 ```
+{% endhint %}
+
+### Weights & Biases UI
+
+Weights & Biases comes with a web-based UI that you can use to find further 
+details about your tracked experiments. 
+
+Every ZenML step that uses Weights & Biases should create a separate experiment 
+run which you can inspect in the Weights & Biases UI:
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/experiment-trackers/wandb.md:175]:
```suggestion
run which you can inspect on the Weights & Biases UI: 
```

File: docs/book/user-guide/component-guide/model-deployers/bentoml.md [hunk 4]
```diff
@@ -62,11 +62,11 @@ zenml model-deployer register bentoml_deployer --flavor=bentoml
 The ZenML integration will provision a local HTTP deployment server as a daemon process that will continue to run in the
 background to serve the latest models and Bentos.
 
-### How do you use it?
+## How do you use it?
 
-In order to use the BentoML Model Deployer, We need to understand these three main concepts:
+In order to use the BentoML Model Deployer, we need to understand three main concepts:
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/model-deployers/bentoml.md:67]:
```suggestion
In order to use the BentoML Model Deployer, you need to have an understanding of the following three concepts:
```

File: docs/book/user-guide/component-guide/step-operators/sagemaker.md [hunk 5]
```diff
@@ -4,8 +4,10 @@ description: Executing individual steps in SageMaker.
 
 # Amazon SageMaker
 
-The SageMaker step operator is a [step operator](step-operators.md) flavor provided with the ZenML `aws` integration
-that uses [SageMaker](https://aws.amazon.com/sagemaker/) to execute individual steps of ZenML pipelines.
+[SageMaker](https://aws.amazon.com/sagemaker/) offers specialized compute 
+instances to run your training jobs and has a beautiful UI to track and manage
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/step-operators/sagemaker.md:8]:
```suggestion
instances to run your training jobs and has a comprehensive UI to track and manage 
```

File: docs/book/user-guide/component-guide/step-operators/azureml.md [hunk 6]
```diff
@@ -4,9 +4,10 @@ description: Executing individual steps in AzureML.
 
 # AzureML
 
-The AzureML step operator is a [step operator](step-operators.md) flavor provided by the ZenML `azure` integration that
-uses [AzureML](https://azure.microsoft.com/en-us/services/machine-learning/) to execute individual steps of ZenML
-pipelines.
+[AzureML](https://azure.microsoft.com/en-us/services/machine-learning/) offers
+specialized compute instances to run your training jobs and has a beautiful UI
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/step-operators/azureml.md:8]:
```suggestion
specialized compute instances to run your training jobs and has a comprehensive UI 
```

File: docs/book/user-guide/component-guide/orchestrators/vertex.md [hunk 7]
```diff
@@ -329,26 +335,45 @@ vertex_settings = VertexOrchestratorSettings(
         ]
     }
 )
+```
+
+If some of your pipelines steps have certain hardware requirements, you can
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/orchestrators/vertex.md:340]:
```suggestion
If your pipelines steps have certain hardware requirements, you can
```

File: docs/book/user-guide/component-guide/orchestrators/tekton.md [hunk 8]
```diff
@@ -233,26 +246,44 @@ tekton_settings = TektonOrchestratorSettings(
         ]
     }
 )
+```
+
+If some of your pipelines steps have certain hardware requirements, you can
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/orchestrators/tekton.md:251]:
```suggestion
If your pipelines steps have certain hardware requirements, you can
```

**Review by bcdurak — CHANGES_REQUESTED** [review]
Amazing work 💯  Hard to believe how many moving parts you had to manage.

I tried covering the docs side first and thought this would be a nice checkpoint to leave some small notes. Feel free to ignore any of them.

File: docs/book/user-guide/advanced-guide/artifact-management/handle-custom-data-types.md [hunk 9]
```diff
@@ -8,10 +8,74 @@ A ZenML pipeline is built in a data-centric way. The outputs and inputs of steps
...
+| [ServiceMaterializer](https://sdkdocs.zenml.io/latest/core_code_docs/core-materializers/#zenml.materializers.service_materializer.ServiceMaterializer) | `zenml.services.service.BaseService` | `.json` |
+| [StructuredStringMaterializer](https://sdkdocs.zenml.io/latest/core_code_docs/core-materializers/#zenml.materializers.structured_string_materializer.StructuredStringMaterializer) | `zenml.types.CSVString`, `zenml.types.HTMLString`, `zenml.types.MarkdownString` | `.csv` / `.html` / `.md` (depending on type) |
+
 {% hint style="info" %}
-ZenML already includes built-in materializers for many common data types. However, if you want to pass custom objects between pipeline steps, these objects are by default saved using [cloudpickle](https://github.com/cloudpipe/cloudpickle) , which is not production-ready because the resulting artifacts cannot be loaded when running with a different Python version. In such cases, you should consider building a custom Materializer to save your objects in a more robust and efficient format.
+ZenML also provides a built-in [CloudpickleMaterializer]()
```
[the remark this answers is not in the document]
Comment (fa9r) [reply docs/book/user-guide/advanced-guide/artifact-management/handle-custom-data-types.md:29]: yeah great catch!

File: docs/book/user-guide/component-guide/experiment-trackers/wandb.md (hunk 3 above)
Comment (fa9r) [reply docs/book/user-guide/component-guide/experiment-trackers/wandb.md:175]: Both of the sentences are grammatically correct, but the first one, "which you can inspect in the Weights & Biases UI," is more commonly used and sounds more natural.

"In the Weights & Biases UI" indicates the location where something can be inspected, so using "in" here is appropriate. The second sentence, "which you can inspect on the Weights & Biases UI," would still be understood, but it might not be as commonly used in this context.

Comment (fa9r) [reply docs/book/user-guide/component-guide/experiment-trackers/wandb.md:175]: Computer says no.

File: docs/book/user-guide/component-guide/step-operators/sagemaker.md (hunk 5 above)
Comment (fa9r) [reply docs/book/user-guide/component-guide/step-operators/sagemaker.md:8]: hahaha fair

File: src/zenml/cli/__init__.py [hunk 10]
```diff
@@ -1596,7 +1596,6 @@ def my_pipeline(...):
 from zenml.cli.base import *  # noqa
 from zenml.cli.code_repository import *  # noqa
 from zenml.cli.config import *  # noqa
-from zenml.cli.example import *  # noqa
```
Comment (bcdurak) [inline src/zenml/cli/__init__.py:1599]: We should remove the `zenml example` documentation from the huge docstring at the top of this file too.

File: docs/book/user-guide/component-guide/model-deployers/bentoml.md (hunk 4 above)
Comment (fa9r) [reply docs/book/user-guide/component-guide/model-deployers/bentoml.md:67]: Good point, the original wording was quite bad here. I rephrased this entirely 👍

File: docs/book/user-guide/component-guide/experiment-trackers/neptune.md (hunk 2 above)
Comment (fa9r) [reply docs/book/user-guide/component-guide/experiment-trackers/neptune.md:175]: Computer says no, see below.

File: docs/book/user-guide/advanced-guide/artifact-management/handle-custom-data-types.md [hunk 11]
```diff
@@ -8,10 +8,74 @@ A ZenML pipeline is built in a data-centric way. The outputs and inputs of steps
...
+However, this is not production-ready because the resulting artifacts cannot be loaded when running with a different Python version. In such cases, you should consider building a [custom Materializer](#custom-materializers) to save your objects in a more robust and efficient format.
+{% endhint %}
+
+## Integration Materializers
+
+In addition to the built-in materializers, ZenML also provides several
+integration-specific materializers that can be activated by installing the
+respective [integration](../../component-guide/integration-overview.md):
+
+| Integration | Materializer | Handled Data Types | Storage Format |
```
[the remark this answers is not in the document]
Comment (fa9r) [reply docs/book/user-guide/advanced-guide/artifact-management/handle-custom-data-types.md:42]: don't know how to do that and a bit scared of editing directly in Gitbook as it will screw up all the headers and assets again

File: src/zenml/cli/__init__.py (hunk 10 above)
Comment (fa9r) [reply src/zenml/cli/__init__.py:1599]: Ah, great catch, almost forgot that I removed `zenml example ...` as part of this PR 😅

Comment (fa9r) [reply src/zenml/cli/__init__.py:1599]: I'll remove all references 👍

File: tests/integration/examples/mlflow/pipelines/tracking_pipeline/tracking_pipeline.py [hunk 12]
```diff
@@ -1,4 +1,4 @@
-#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#  Copyright (c) ZenML GmbH 2020. All Rights Reserved.
```
Comment (bcdurak) [inline tests/integration/examples/mlflow/pipelines/tracking_pipeline/tracking_pipeline.py:1]: Any reason for this update? Shouldn't it be 2023? 😄

Comment (fa9r) [reply tests/integration/examples/mlflow/pipelines/tracking_pipeline/tracking_pipeline.py:1]: didn't change it, this was the kubeflow example pipeline before
